core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -72,13 +71,13 @@ private[spark] class CoarseMesosSchedulerBackend(
   private val shuffleServiceEnabled = conf.getBoolean("spark.shuffle.service.enabled", false)
 
   // Cores we have acquired with each Mesos task ID
-  val coresByTaskId = new HashMap[String, Int]
+  val coresByTaskId = new mutable.HashMap[String, Int]
```

Comment:
Right now the code has at the imports section:

import scala.collection.mutable
import scala.collection.mutable.{Buffer, HashMap, HashSet}

and bellow:
val coresByTaskId = new HashMap[String, Int]

So the mutable is used anyway, at least this is shown to my IDE.

I just made the mutable explicit because its cleaner.

As for the mutable vs immutable problem mutable is already used and IMHO needed eg.
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala#L483